internal/crypto/ptls/dialer.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package ptls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"net"
+	"time"
+)
+
+type Dialer interface {
+	IsReachableAndTLSValidationSucceeds(
```

Comment:
suggest adding a new parameter to take in a `context` that can then be used to `tls.DialContext` and do something like
```

	foo := tls.Dialer{
		Config: ptls.Default(certPool),
	}
	conn, err := foo.DialContext(ctx, "tcp", endpointHostPort.Endpoint())
```

If the incoming context is `nil`, we can pass a context w/ a default timeout, of say 30s